fem/gridfunc.cpp

```diff
@@ -1236,6 +1236,44 @@ void GridFunction::AccumulateAndCountZones(Coefficient &coeff,
    }
 }
 
+void GridFunction::AccumulateAndCountZones(VectorCoefficient &vcoeff,
+                                           AvgType type,
+                                           Array<int> &zones_per_vdof)
+{
+   zones_per_vdof.SetSize(fes->GetVSize());
+   zones_per_vdof = 0;
+
+   // Local interpolation
+   Array<int> vdofs;
+   Vector vals;
+   *this = 0.0;
+   for (int i = 0; i < fes->GetNE(); i++)
+   {
+      fes->GetElementVDofs(i, vdofs);
+      // Local interpolation of coeff.
+      vals.SetSize(vdofs.Size());
+      fes->GetFE(i)->Project(vcoeff, *fes->GetElementTransformation(i), vals);
+
```

Comment:
In the other version of the function, that uses Coefficient &coeff, can't we just cast it:
```
VectorCoefficient *vecc = dynamic_cast<VectorCoefficient *>(&coeff);
if (vecc) { fes->GetFE(i)->Project(*vecc, *fes->GetElementTransformation(i), vals); }
else      { fes->GetFE(i)->Project(coeff, *fes->GetElementTransformation(i), vals); }
```

Reply:
At the moment, `VectorCoefficient` does not inherit from `Coefficient`.